lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -1960,6 +1960,52 @@ def test__lazy(self):
         self._check_copy(cube, cube.copy())
 
 
+def _add_test_meshcube(self, nomesh=False, n_z=2, **meshcoord_kwargs):
+    # A common setup action : Create a standard test cube with a variety of
+    # types of coord, and add various objects to the testcase,
+    # i.e. to "self".
+    if nomesh:
+        mesh = None
+        n_faces = 5
+    else:
+        mesh = meshcoord_kwargs.pop("mesh", None)
+        if mesh is None:
+            mesh = create_test_mesh()
+        meshx, meshy = (
+            create_test_meshcoord(axis=axis, mesh=mesh, **meshcoord_kwargs)
+            for axis in ("x", "y")
+        )
+        n_faces = meshx.shape[0]
+
+    mesh_dimco = DimCoord(
+        np.arange(n_faces), long_name="i_mesh_face", units="1"
+    )
+
+    auxco_x = AuxCoord(np.zeros(n_faces), long_name="mesh_face_aux", units="1")
+
+    zco = DimCoord(np.arange(n_z), long_name="level", units=1)
+    cube = Cube(np.zeros((n_z, n_faces)), long_name="mesh_phenom")
+    cube.add_dim_coord(zco, 0)
+    if nomesh:
+        mesh_coords = []
+    else:
+        mesh_coords = [meshx, meshy]
+
+    cube.add_dim_coord(mesh_dimco, 1)
+    for co in mesh_coords + [auxco_x]:
+        cube.add_aux_coord(co, 1)
+
+    self.dimco_z = zco
+    self.dimco_mesh = mesh_dimco
+    if not nomesh:
+        self.meshco_x = meshx
+        self.meshco_y = meshy
+    self.auxco_x = auxco_x
+    self.allcoords = mesh_coords + [zco, mesh_dimco, auxco_x]
+    self.cube = cube
+    self.mesh = mesh
+
+
```

Comment:
@pp-mo If we're serious about supporting unstructured meshes and cube-spheres... and I think we are... would you consider us acknowledge this by curating some `iris.tests.stock` cubes dedicated to testing `iris` with synthetic based cubes such as this?

I seems to me that this kind of cube generation has value beyond this specific pull-request. At least we could start that commitment here.

So if you buy into that point-of-view, fancy migrating this into a separate stock cube namespace, say `iris.tests.stock._mesh`, and then update `iris.tests.stock` to `from ._mesh import <whatever>` ?

Reply:
We are already committed to synthetic data generation as part of MPerf testing 👍

Reply:
> synthetic data generation
We are, but I think that all goes through *files* at this point ?
So we could maybe still use a means of generating stock *cubes*.

I will look + see if we can maybe integrate the two, though : Perhaps the same code, with the same options, can be used to generate either files or cubes.
)Obviously, save capability is still some way off)